src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -15,22 +15,123 @@
+
+        recommendationRequestRepository.delete(recommendationRequest);
+
+        return genericMessage("RecommendationRequest with id %s deleted".formatted(id));
+    }
+
+    /**
+     * The user who posted a RecommendationRequest can delete their RecommendationRequest
+     * 
+     * @param id the id of the RecommendationRequest to delete
+     * @return a message indicating that the RecommendationRequest was deleted
+     */
+    @Operation(summary = "User can delete their RecommendationRequest")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @DeleteMapping("")
+    public Object deleteRecommendationRequestAsUser(@Parameter(name = "id") @RequestParam Long id) {
+        User currentUser = getCurrentUser().getUser(); 
+        RecommendationRequest recommendationRequest = 
+        recommendationRequestRepository
+            .findByIdAndRequester(id, currentUser)
+            .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        recommendationRequestRepository.delete(recommendationRequest);
+
+        return genericMessage("RecommendationRequest with id %s deleted".formatted(id));
+    }
+
+    /**
+     * The user who posted a RecommendationRequest can update their RecommendationRequest
+     * 
+     * @param id       the id of the Recommendation Request to update
+     * @param incoming the updated Recommendation Request
+     * @return the updated Recommendation Request object
+     */
+    @Operation(summary = "User can update their RecommendationRequest")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PutMapping("")
+    public RecommendationRequest updateRecommendationRequestAsUser( 
+        @Parameter(name = "id") @RequestParam Long id,
+        @RequestBody @Valid RecommendationRequest incoming) {
+
+        User currentUser = getCurrentUser().getUser(); 
+        RecommendationRequest recommendationRequest = 
+            recommendationRequestRepository
+                .findByIdAndRequester(id, currentUser)
+                .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        recommendationRequest.setDetails(incoming.getDetails());
+
+        recommendationRequestRepository.save(recommendationRequest);
+           
+        return recommendationRequest;    
+    }
+
+     /**
+     * Prof can update a Recommendation Request's status
+     * 
+     * @param id       the id of the Recommendation Request to update
+     * @param incoming the updated Recommendation Request
+     * @return the updated Recommendation Request object
+     */
+    @Operation(summary = "A Professor can update a recommendation request's status")
+    @PreAuthorize("hasRole('ROLE_PROFESSOR')")
+    @PutMapping("/professor")
+    public RecommendationRequest updateRecommendationRequestAsAdmin(
+        @Parameter(name = "id") @RequestParam Long id,
+        @RequestBody @Valid RecommendationRequest incoming) {
```

Comment:
For the professor put, I think instead of the entire body being editable, it should only have one area to fill in which is the status - professors should be able to modify the status but nothing more

Reply:
I thought that by only adding 'recommendationRequest.setStatus(incoming.getStatus());', they would only be allowed to update the status.